cli_tools/import_precheck/check_disks_linux.go

```diff
@@ -60,47 +87,114 @@ func (c *disksCheck) getName() string {
 	return "Disks Check"
 }
 
-func (c *disksCheck) run() (*report, error) {
-	r := &report{name: c.getName()}
-	var out string
-	var err error
-	if c.lsblkOverride != nil {
-		out, err = c.lsblkOverride()
-	} else {
-		out, err = c.lsblk()
+// mountPoints supports listing the mounts on a system, and determining which block device(s)
+// the mount is physically contained. This is helpful since technologies such as LVM allow
+// logical mounts to span physical devices.
+type mountPoints struct {
+	mounts []struct {
+		dir       string
+		hierarchy []string
 	}
-	if err != nil {
-		return nil, err
+}
+
+// listPhysicalDevicesForMount returns the the block device(s) that contain the mount.
+// An empty return value means the mount was not found.
+func (m *mountPoints) listPhysicalDevicesForMount(mount string) (devices []string) {
+	set := map[string]struct{}{}
+	for _, mp := range m.mounts {
+		if mp.dir == mount {
+			set[mp.hierarchy[0]] = struct{}{}
+		}
 	}
+	for k := range set {
+		devices = append(devices, k)
+	}
+	// sorted for stability in testing.
+	sort.Strings(devices)
+	return devices
+}
 
-	l := lsblkParse(out)
+// listMountPoints returns all mounts.
+func (m *mountPoints) listMountPoints() (mounts []string) {
+	set := map[string]struct{}{}
+	for _, mount := range m.mounts {
+		set[mount.dir] = struct{}{}
+	}
+	for k := range set {
+		mounts = append(mounts, k)
+	}
+	// sorted for stability in testing.
+	sort.Strings(mounts)
+	return mounts
+}
 
-	r.Info(fmt.Sprintf("`lsblk -i` results:\n%s", l.raw))
-	if len(l.devRows) != 1 {
-		r.Warn("translation only supports single block device. Will attempt to determine if translation is possible...")
+// addAll populates the mountPoints data structure with the response from lsblk.
+func (m *mountPoints) addAll(elements []DiskElement, basePath []string) {
+	for _, element := range elements {
+		path := make([]string, len(basePath))
+		copy(path, basePath)
+		path = append(path, element.Name)
+		if element.Mountpoint != "" {
+			m.mounts = append(m.mounts, struct {
+				dir       string
+				hierarchy []string
+			}{element.Mountpoint, path})
```

Comment:
Maybe have this struct as a type, since it's repeated here and in the definition of mountPoints

Reply:
done